### ISSUE TITLE
Unescaped } breaks expansions that follow it

### ISSUE THREAD

**Issue by hexagonrecursion** [tox-dev/tox#1956]
Based on the below test it looks like escaping only the opening or only the closing bracket is allowed. I.e. `\{posargs}` is valid syntax and expands to literal `{posargs}`.
https://github.com/tox-dev/tox/blob/651e1ae9203a095f422c48c9673e60696f8650c1/tests/config/loader/ini/replace/test_replace_posargs.py#L45-L58

I have found a case where this syntax breaks expansions that follow it:
```
\{posargs} {posargs}
```

# To reproduce
## tox.ini
```ini
[tox]
no_package = True

[testenv]
commands = python -c 'import sys; print(sys.argv)' \{posargs} {posargs}
```
## Run
```
tox4 -- foo
```
## Expected output:
```
['-c', '{posargs}', 'foo']
```
## Actual output:
```
['-c', '{posargs}', '{posargs}']
```
## tox4 -rvv -- foo
<details>
<summary> Collapsed for brevity </summary>

```
ROOT: 407 D setup logging to DEBUG on pid 15093 [tox/report.py:211]
ROOT: 417 W remove tox env folder /home/user/hello-tox/.tox/4/py [tox/tox_env/api.py:202]
ROOT: 533 I find interpreter for spec PythonSpec(implementation=CPython) [virtualenv/discovery/builtin.py:62]
ROOT: 533 D discover exe for PythonInfo(spec=CPython3.9.1.final.0-64, exe=/home/user/path/to/tox4-venv/bin/python, platform=linux, version='3.9.1 (default, Dec 11 2020, 14:32:07) \n[GCC 7.3.0]', encoding_fs_io=utf-8-utf-8) in /home/user/.conda/envs/py3 [virtualenv/discovery/py_info.py:372]
ROOT: 535 D filesystem is case-sensitive [virtualenv/info.py:28]
ROOT: 537 D Attempting to acquire lock 135013968838864 on /home/user/.local/share/virtualenv/py_info/1/32916b286f4a70af59e238bfd51ed56d548abf51ddd087759ce7be39e4509643.lock [filelock.py:270]
ROOT: 537 D Lock 135013968838864 acquired on /home/user/.local/share/virtualenv/py_info/1/32916b286f4a70af59e238bfd51ed56d548abf51ddd087759ce7be39e4509643.lock [filelock.py:274]
ROOT: 538 D got python info of /home/user/.conda/envs/py3/bin/python3.9 from /home/user/.local/share/virtualenv/py_info/1/32916b286f4a70af59e238bfd51ed56d548abf51ddd087759ce7be39e4509643.json [virtualenv/app_data/via_disk_folder.py:135]
ROOT: 540 D Attempting to release lock 135013968838864 on /home/user/.local/share/virtualenv/py_info/1/32916b286f4a70af59e238bfd51ed56d548abf51ddd087759ce7be39e4509643.lock [filelock.py:315]
ROOT: 540 D Lock 135013968838864 released on /home/user/.local/share/virtualenv/py_info/1/32916b286f4a70af59e238bfd51ed56d548abf51ddd087759ce7be39e4509643.lock [filelock.py:318]
ROOT: 540 I proposed PythonInfo(spec=CPython3.9.1.final.0-64, system=/home/user/.conda/envs/py3/bin/python3.9, exe=/home/user/path/to/tox4-venv/bin/python, platform=linux, version='3.9.1 (default, Dec 11 2020, 14:32:07) \n[GCC 7.3.0]', encoding_fs_io=utf-8-utf-8) [virtualenv/discovery/builtin.py:69]
ROOT: 541 D accepted PythonInfo(spec=CPython3.9.1.final.0-64, system=/home/user/.conda/envs/py3/bin/python3.9, exe=/home/user/path/to/tox4-venv/bin/python, platform=linux, version='3.9.1 (default, Dec 11 2020, 14:32:07) \n[GCC 7.3.0]', encoding_fs_io=utf-8-utf-8) [virtualenv/discovery/builtin.py:71]
ROOT: 628 I create virtual environment via CPython3Posix(dest=/home/user/hello-tox/.tox/4/py, clear=True, no_vcs_ignore=False, global=False) [virtualenv/run/session.py:52]
ROOT: 628 D delete /home/user/hello-tox/.tox/4/py [virtualenv/create/creator.py:169]
ROOT: 629 D create folder /home/user/hello-tox/.tox/4/py/bin [virtualenv/util/path/_sync.py:25]
ROOT: 629 D create folder /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages [virtualenv/util/path/_sync.py:25]
ROOT: 630 D write /home/user/hello-tox/.tox/4/py/pyvenv.cfg [virtualenv/create/pyenv_cfg.py:34]
ROOT: 630 D 	home = /home/user/.conda/envs/py3 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 630 D 	implementation = CPython [virtualenv/create/pyenv_cfg.py:38]
ROOT: 630 D 	version_info = 3.9.1.final.0 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 630 D 	virtualenv = 20.4.2 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 630 D 	include-system-site-packages = false [virtualenv/create/pyenv_cfg.py:38]
ROOT: 630 D 	base-prefix = /home/user/.conda/envs/py3 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 630 D 	base-exec-prefix = /home/user/.conda/envs/py3 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 631 D 	base-executable = /home/user/.conda/envs/py3/bin/python3.9 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 631 D symlink /home/user/.conda/envs/py3/bin/python3.9 to /home/user/hello-tox/.tox/4/py/bin/python [virtualenv/util/path/_sync.py:44]
ROOT: 633 D create virtualenv import hook file /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/_virtualenv.pth [virtualenv/create/via_global_ref/api.py:95]
ROOT: 633 D create /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/_virtualenv.py [virtualenv/create/via_global_ref/api.py:98]
ROOT: 634 D ============================== target debug ============================== [virtualenv/run/session.py:54]
ROOT: 635 D debug via /home/user/hello-tox/.tox/4/py/bin/python /home/user/path/to/tox4-venv/lib/python3.9/site-packages/virtualenv/create/debug.py [virtualenv/create/creator.py:224]
ROOT: 634 D {
  "sys": {
    "executable": "/home/user/hello-tox/.tox/4/py/bin/python",
    "_base_executable": "/home/user/hello-tox/.tox/4/py/bin/python",
    "prefix": "/home/user/hello-tox/.tox/4/py",
    "base_prefix": "/home/user/.conda/envs/py3",
    "real_prefix": null,
    "exec_prefix": "/home/user/hello-tox/.tox/4/py",
    "base_exec_prefix": "/home/user/.conda/envs/py3",
    "path": [
      "/home/user/.conda/envs/py3/lib/python39.zip",
      "/home/user/.conda/envs/py3/lib/python3.9",
      "/home/user/.conda/envs/py3/lib/python3.9/lib-dynload",
      "/home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages"
    ],
    "meta_path": [
      "<class '_virtualenv._Finder'>",
      "<class '_frozen_importlib.BuiltinImporter'>",
      "<class '_frozen_importlib.FrozenImporter'>",
      "<class '_frozen_importlib_external.PathFinder'>"
    ],
    "fs_encoding": "utf-8",
    "io_encoding": "utf-8"
  },
  "version": "3.9.1 (default, Dec 11 2020, 14:32:07) \n[GCC 7.3.0]",
  "makefile_filename": "/home/user/.conda/envs/py3/lib/python3.9/config-3.9-x86_64-linux-gnu/Makefile",
  "os": "<module 'os' from '/home/user/.conda/envs/py3/lib/python3.9/os.py'>",
  "site": "<module 'site' from '/home/user/.conda/envs/py3/lib/python3.9/site.py'>",
  "datetime": "<module 'datetime' from '/home/user/.conda/envs/py3/lib/python3.9/datetime.py'>",
  "math": "<module 'math' from '/home/user/.conda/envs/py3/lib/python3.9/lib-dynload/math.cpython-39-x86_64-linux-gnu.so'>",
  "json": "<module 'json' from '/home/user/.conda/envs/py3/lib/python3.9/json/__init__.py'>"
} [virtualenv/run/session.py:55]
ROOT: 708 I add seed packages via FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/user/.local/share/virtualenv) [virtualenv/run/session.py:59]
ROOT: 719 D got embed update of distribution wheel from /home/user/.local/share/virtualenv/wheel/3.9/embed/1/wheel.json [virtualenv/app_data/via_disk_folder.py:135]
ROOT: 721 D got embed update of distribution setuptools from /home/user/.local/share/virtualenv/wheel/3.9/embed/1/setuptools.json [virtualenv/app_data/via_disk_folder.py:135]
ROOT: 722 D got embed update of distribution pip from /home/user/.local/share/virtualenv/wheel/3.9/embed/1/pip.json [virtualenv/app_data/via_disk_folder.py:135]
ROOT: 728 D using periodically updated wheel /home/user/.local/share/virtualenv/wheel/house/setuptools-53.0.0-py3-none-any.whl [virtualenv/seed/wheels/periodic_update.py:53]
ROOT: 729 D install wheel from wheel /home/user/path/to/tox4-venv/lib/python3.9/site-packages/virtualenv/seed/wheels/embed/wheel-0.36.2-py2.py3-none-any.whl via CopyPipInstall [virtualenv/seed/embed/via_app_data/via_app_data.py:49]
ROOT: 729 D install pip from wheel /home/user/path/to/tox4-venv/lib/python3.9/site-packages/virtualenv/seed/wheels/embed/pip-21.0.1-py3-none-any.whl via CopyPipInstall [virtualenv/seed/embed/via_app_data/via_app_data.py:49]
ROOT: 730 D install setuptools from wheel /home/user/.local/share/virtualenv/wheel/house/setuptools-53.0.0-py3-none-any.whl via CopyPipInstall [virtualenv/seed/embed/via_app_data/via_app_data.py:49]
ROOT: 732 D Attempting to acquire lock 135013969084128 on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/pip-21.0.1-py3-none-any.lock [filelock.py:270]
ROOT: 733 D Attempting to acquire lock 135013941858064 on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any.lock [filelock.py:270]
ROOT: 734 D Lock 135013969084128 acquired on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/pip-21.0.1-py3-none-any.lock [filelock.py:274]
ROOT: 734 D Attempting to acquire lock 135013941855472 on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/wheel-0.36.2-py2.py3-none-any.lock [filelock.py:270]
ROOT: 736 D Lock 135013941858064 acquired on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any.lock [filelock.py:274]
ROOT: 736 D Lock 135013941855472 acquired on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/wheel-0.36.2-py2.py3-none-any.lock [filelock.py:274]
ROOT: 738 D Attempting to release lock 135013969084128 on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/pip-21.0.1-py3-none-any.lock [filelock.py:315]
ROOT: 738 D Lock 135013969084128 released on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/pip-21.0.1-py3-none-any.lock [filelock.py:318]
ROOT: 738 D Attempting to release lock 135013941855472 on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/wheel-0.36.2-py2.py3-none-any.lock [filelock.py:315]
ROOT: 739 D Lock 135013941855472 released on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/wheel-0.36.2-py2.py3-none-any.lock [filelock.py:318]
ROOT: 739 D Attempting to release lock 135013941858064 on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any.lock [filelock.py:315]
ROOT: 740 D Lock 135013941858064 released on /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any.lock [filelock.py:318]
ROOT: 741 D copy /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/pip-21.0.1-py3-none-any/pip-21.0.1.virtualenv to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/pip-21.0.1.virtualenv [virtualenv/util/path/_sync.py:52]
ROOT: 741 D copy directory /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/wheel-0.36.2-py2.py3-none-any/wheel-0.36.2.dist-info to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/wheel-0.36.2.dist-info [virtualenv/util/path/_sync.py:52]
ROOT: 741 D copy directory /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any/_distutils_hack to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/_distutils_hack [virtualenv/util/path/_sync.py:52]
ROOT: 743 D copy directory /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/pip-21.0.1-py3-none-any/pip to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/pip [virtualenv/util/path/_sync.py:52]
ROOT: 751 D copy directory /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any/setuptools to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/setuptools [virtualenv/util/path/_sync.py:52]
ROOT: 768 D copy directory /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/wheel-0.36.2-py2.py3-none-any/wheel to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/wheel [virtualenv/util/path/_sync.py:52]
ROOT: 850 D copy /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/wheel-0.36.2-py2.py3-none-any/wheel-0.36.2.virtualenv to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/wheel-0.36.2.virtualenv [virtualenv/util/path/_sync.py:52]
ROOT: 855 D changing mode of /home/user/hello-tox/.tox/4/py/bin/wheel-3.9 to 775 [distlib/util.py:566]
ROOT: 857 D changing mode of /home/user/hello-tox/.tox/4/py/bin/wheel3 to 775 [distlib/util.py:566]
ROOT: 858 D changing mode of /home/user/hello-tox/.tox/4/py/bin/wheel to 775 [distlib/util.py:566]
ROOT: 861 D changing mode of /home/user/hello-tox/.tox/4/py/bin/wheel3.9 to 775 [distlib/util.py:566]
ROOT: 861 D generated console scripts wheel-3.9 wheel3 wheel3.9 wheel [virtualenv/seed/embed/via_app_data/pip_install/base.py:48]
ROOT: 1018 D copy directory /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any/setuptools-53.0.0.dist-info to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/setuptools-53.0.0.dist-info [virtualenv/util/path/_sync.py:52]
ROOT: 1031 D copy directory /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any/pkg_resources to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/pkg_resources [virtualenv/util/path/_sync.py:52]
ROOT: 1109 D copy /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any/distutils-precedence.pth to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/distutils-precedence.pth [virtualenv/util/path/_sync.py:52]
ROOT: 1113 D copy /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-53.0.0-py3-none-any/setuptools-53.0.0.virtualenv to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/setuptools-53.0.0.virtualenv [virtualenv/util/path/_sync.py:52]
ROOT: 1115 D generated console scripts  [virtualenv/seed/embed/via_app_data/pip_install/base.py:48]
ROOT: 1390 D copy directory /home/user/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/pip-21.0.1-py3-none-any/pip-21.0.1.dist-info to /home/user/hello-tox/.tox/4/py/lib/python3.9/site-packages/pip-21.0.1.dist-info [virtualenv/util/path/_sync.py:52]
ROOT: 1398 D changing mode of /home/user/hello-tox/.tox/4/py/bin/pip3.9 to 775 [distlib/util.py:566]
ROOT: 1398 D changing mode of /home/user/hello-tox/.tox/4/py/bin/pip-3.9 to 775 [distlib/util.py:566]
ROOT: 1399 D changing mode of /home/user/hello-tox/.tox/4/py/bin/pip to 775 [distlib/util.py:566]
ROOT: 1399 D changing mode of /home/user/hello-tox/.tox/4/py/bin/pip3 to 775 [distlib/util.py:566]
ROOT: 1399 D generated console scripts pip pip-3.9 pip3 pip3.9 [virtualenv/seed/embed/via_app_data/pip_install/base.py:48]
ROOT: 1400 I add activators for Bash, CShell, Fish, PowerShell, Python, Xonsh [virtualenv/run/session.py:64]
ROOT: 1408 D write /home/user/hello-tox/.tox/4/py/pyvenv.cfg [virtualenv/create/pyenv_cfg.py:34]
ROOT: 1408 D 	home = /home/user/.conda/envs/py3 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 1409 D 	implementation = CPython [virtualenv/create/pyenv_cfg.py:38]
ROOT: 1409 D 	version_info = 3.9.1.final.0 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 1409 D 	virtualenv = 20.4.2 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 1409 D 	include-system-site-packages = false [virtualenv/create/pyenv_cfg.py:38]
ROOT: 1409 D 	base-prefix = /home/user/.conda/envs/py3 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 1409 D 	base-exec-prefix = /home/user/.conda/envs/py3 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 1409 D 	base-executable = /home/user/.conda/envs/py3/bin/python3.9 [virtualenv/create/pyenv_cfg.py:38]
ROOT: 1413 W commands[0]> python -c 'import sys; print(sys.argv)' '{posargs}' '{posargs}' [tox/tox_env/api.py:288]
['-c', '{posargs}', '{posargs}']
ROOT: 1474 I exit 0 (0.06 seconds) /home/user/hello-tox> python -c 'import sys; print(sys.argv)' '{posargs}' '{posargs}' pid=15117 [tox/execute/api.py:193]
  py: OK (1.06=setup[1.00]+cmd[0.06] seconds)
  congratulations :) (1.13 seconds)
```

</details>

**Comment by gaborbernat** [comment]
> ## setup.py
> ```python
> from distutils.core import setup
> setup()
> ```

Please don't use distutils, it's deprecated. For the point of this test you don't need any packaging, you can just set skip_packaging 👍🏻

**Comment by hexagonrecursion** [comment]
>`skip_packaging`

Did you mean `[testenv] skip_install = true` ?

**Comment by gaborbernat** [comment]
Sorry I meant ``no_package`` - https://github.com/tox-dev/tox/blob/rewrite/src/tox/tox_env/runner.py#L118

**Comment by hexagonrecursion** [comment]
Fixed by #1961
